buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -120,28 +128,28 @@ object Java9Modularity {
         val kotlinApiPlugin = plugins.getPlugin(KotlinApiPlugin::class)
         val verifyModuleTask = kotlinApiPlugin.registerKotlinJvmCompileTask(
             verifyModuleTaskName,
-            compileTask.compilerOptions.moduleName.get()
+            compilerOptions = compileTask.get().compilerOptions,
+            explicitApiMode = provider { ExplicitApiMode.Disabled }
         )
         verifyModuleTask {
             group = VERIFICATION_GROUP
             description = "Verify Kotlin sources for JPMS problems"
-            libraries.from(compileTask.libraries)
-            source(compileTask.sources)
-            source(compileTask.javaSources)
+            libraries.from(compileTask.map { it.libraries })
+            source(compileTask.map { it.sources })
+            source(compileTask.map { it.javaSources })
             // part of work-around for https://youtrack.jetbrains.com/issue/KT-60541
-            @Suppress("INVISIBLE_MEMBER")
-            source(compileTask.scriptSources)
+            source(compileTask.map {
+                @Suppress("INVISIBLE_MEMBER")
+                it.scriptSources
+            })
             source(sourceFile)
             destinationDirectory.set(temporaryDir)
-            multiPlatformEnabled.set(compileTask.multiPlatformEnabled)
+            multiPlatformEnabled.set(compileTask.get().multiPlatformEnabled)
             compilerOptions {
                 jvmTarget.set(JvmTarget.JVM_9)
-                // To support LV override when set in aggregate builds
-                languageVersion.set(compileTask.compilerOptions.languageVersion)
```

Comment:
Why did you remove LV + optIns copying? IIRC lack of proper LV caused some problems in aggregate/user projects builds.

Reply:
On line 139 in the new code the whole `compilerOptions` object from relevant compile task are used as a base for this task. This should also include proper LV + optIns.